File: vps-server/src/main/java/com/vps/vpsserver/entity/Order.java
package com.vps.vpsserver.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "price_group_id", nullable = false)
    private PriceGroup priceGroup;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "server_id")
    private Server server;

    @Column(name = "order_number", unique = true, nullable = false)
    private String orderNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status = OrderStatus.PENDING;

    @Column(nullable = false, precision = 19, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false, length = 3)
    private String currency = "CNY";

    @Column(name = "billing_period", nullable = false)
    private String billingPeriod; // hourly, daily, monthly, quarterly, semiAnnual, annual

    // 服务器配置信息
    @Column(name = "cpu_cores")
    private Integer cpuCores;

    @Column(name = "memory_gb")
    private Integer memoryGb;

    @Column(name = "storage_gb")
    private Integer storageGb;

    @Column(name = "bandwidth_mbps")
    private Integer bandwidthMbps;

    @Column(name = "ip_count")
    private Integer ipCount;

    @Column(name = "os_name")
    private String osName;

    @Column(name = "os_version")
    private String osVersion;

    @Column(name = "initial_password")
    private String initialPassword;

    @Column(name = "ssh_port")
    private Integer sshPort;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (orderNumber == null) {
            orderNumber = generateOrderNumber();
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    private String generateOrderNumber() {
        return "ORD" + System.currentTimeMillis() + String.format("%04d", (int)(Math.random() * 10000));
    }

    public enum OrderStatus {
        PENDING,     // 待处理
        PAID,        // 已支付
        PROCESSING,  // 处理中
        ACTIVE,      // 已激活
        SUSPENDED,   // 已暂停
        CANCELLED,   // 已取消
        EXPIRED      // 已过期
    }
}

File: vps-web/src/api/order.ts
import { apiClient } from './client'

export interface CreateOrderRequest {
  priceGroupId: number
  billingPeriod: string
  cpuCores: number
  memoryGb: number
  storageGb: number
  bandwidthMbps: number
  ipCount: number
  osName: string
  osVersion: string
  initialPassword: string
  sshPort: number
  autoRenewal?: boolean
}

export interface OrderDTO {
  id: number
  userId: number
  username: string
  priceGroupId: number
  priceGroupName: string
  serverId?: number
  serverName?: string
  orderNumber: string
  status: 'PENDING' | 'PAID' | 'PROCESSING' | 'ACTIVE' | 'SUSPENDED' | 'CANCELLED' | 'EXPIRED'
  amount: number
  currency: string
  billingPeriod: string
  cpuCores: number
  memoryGb: number
  storageGb: number
  bandwidthMbps: number
  ipCount: number
  osName: string
  osVersion: string
  initialPassword: string
  sshPort: number
  createdAt: string
  updatedAt: string
  expiresAt: string
  autoRenewal?: boolean
  // 服务器管理相关字段
  ipAddress?: string
}

export const orderApi = {
  // 创建订单
  createOrder: async (data: CreateOrderRequest) => {
    const response = await apiClient.post('/orders', data)
    return response.data
  },

  // 获取用户订单列表
  getUserOrders: async () => {
    const response = await apiClient.get('/orders')
    return response.data
  },

  // 根据ID获取订单
  getOrderById: async (id: number) => {
    const response = await apiClient.get(`/orders/${id}`)
    return response.data
  },

  // 根据订单号获取订单
  getOrderByNumber: async (orderNumber: string) => {
    const response = await apiClient.get(`/orders/number/${orderNumber}`)
    return response.data
  },

  // 更新订单状态
  updateOrderStatus: async (id: number, status: string) => {
    const response = await apiClient.put(`/orders/${id}/status?status=${status}`)
    return response.data
  },

  // 取消订单
  cancelOrder: async (id: number) => {
    const response = await apiClient.put(`/orders/${id}/cancel`)
    return response.data
  },

  // 处理订单支付
  processOrderPayment: async (id: number) => {
    const response = await apiClient.post(`/orders/${id}/pay`)
    return response.data
  },

  // 更新订单自动续费设置
  updateAutoRenewal: async (id: number, autoRenewal: boolean) => {
    const response = await apiClient.put(`/orders/${id}/auto-renewal`, { autoRenewal })
    return response.data
  }
}
